crates/full-node/sov-ledger-apis/src/lib.rs

```diff
@@ -862,20 +876,33 @@ struct Batch<B, TxReceipt: TxReceiptContents, E> {
     pub hash: HexHash,
     pub tx_range: Range<u64>,
     pub receipt: B,
-    pub txs: Vec<Transaction<TxReceipt, E>>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub txs: Option<Vec<Transaction<TxReceipt, E>>>,
     pub slot_number: SlotNumber,
 }
 
 impl<B, TxReceipt: TxReceiptContents, E> Batch<B, TxReceipt, E> {
     fn new(batch: BatchResponse<B, TxReceipt, RuntimeEventResponse<E>>, number: u64) -> Self {
-        let mut txs = vec![];
-
-        for tx_response in batch.txs.unwrap_or_default().into_iter() {
-            if let ItemOrHash::Full(tx) = tx_response {
-                txs.push(Transaction::new(tx, number));
+        let txs = match batch.txs {
+            Some(txs) => {
+                let mut output = vec![];
+                let mut failed_conversion = false;
+                for (offset, tx) in txs.into_iter().enumerate() {
+                    if let ItemOrHash::Full(tx) = tx {
+                        output.push(Transaction::new(tx, batch.tx_range.start + offset as u64));
+                    } else {
+                        failed_conversion = true;
+                        break;
+                    }
+                }
+                if failed_conversion {
```

Comment:
In which case conversion is going to fail? Should we log it? Or is it logged inside ::new?

Reply:
This would fail if you did querymode::compact. But in that case, returning nothing matches the semantics of the API as closely as possible, so I think this is ok